src/Model/Config/Communication.php

```diff
@@ -128,29 +128,29 @@ protected function getChannel(string $langAbbr): string
 
     private function ngPath(array $categories, string $param): string
     {
-        $categoryPath = implode(urlencode('/'), array_map(function ($category) {
-            return $this->plusDoubleEncode($category);
-        }, array_reverse($categories)));
+        $categoryPath = array_map(function ($category) {
+            return (string) $this->encodeCategoryName(trim($category));
+        }, array_reverse($categories));
 
-        return sprintf('filter=%s%s', urlencode("$param:"), $categoryPath);
+        return sprintf('filter=%s', urlencode($param . ':' . implode('/', $categoryPath)));
     }
 
     private function standardPath(array $categories, string $param): string
     {
-        $categoriesReverse = array_reverse($categories);
         $path              = 'ROOT';
         $value             = ['navigation=true'];
-        foreach ($categoriesReverse as $key => $category) {
-            $path .= $key === 0 ? null : urlencode('/') . $this->plusDoubleEncode($categoriesReverse[$key - 1]);
-            $parameterKey = urlencode($param) . $path;
-            $value[]      = sprintf('filter%s=%s', $parameterKey, urlencode($category));
+        foreach (array_reverse($categories) as $category) {
+            $value[] = sprintf("filter{$param}%s=%s", $path, urlencode(trim($category)));
+            $path .= urlencode('/' . $this->encodeCategoryName(trim($category)));
         }
 
         return implode(',', $value);
     }
 
-    private function plusDoubleEncode(string $path): string
+    private function encodeCategoryName(string $path): string
     {
-        return urlencode(str_replace('+', ' ', urlencode($path)));
+        //important! do not override this method
```

Comment:
maybe we can use here a **final** keyword to prevent overriding method  **encodeCategoryName**?

Reply:
i think private final method  is rather redundant, as you cannot override the method in a child class.
The message here is just to inform that any modification of that code might be dangerous.
What do you think would be good to put here?

Reply:
Yes, you write, there is no sence to declare private method as final, in PHP8 it will cause even a warning. I guess in this case this message, which you provide, should be enough, I cant see any other option to prevent overriding for private method